packages/contracts/src/contracts/current/protocol/Exchange/MixinWrapperFunctions.sol

```diff
@@ -225,7 +321,7 @@ contract MixinWrapperFunctions is
         returns (FillResults memory fillResults)
     {
         for (uint256 i = 0; i < orders.length; i++) {
-            require(orders[i].takerTokenAddress == orders[0].takerTokenAddress);
+            require(areBytesEqual(orders[i].takerAssetData, orders[0].takerAssetData));
```

Comment:
Hmm, this is tricky. It's possible that someone wants to trade the same token using different proxies, in which case this `require` would fail. This will also be much more expensive than the old `require`. I think we can remove these completely and document that that they should only be used with orders for the same takerToken (for marketSells) or makerToken (for marketBuys). This can always be determined off-chain, but the requires were nice for preventing human error.

Reply:
I would vote for keeping this assertion in on chain as it prevents human error (and accidental purchase of the wrong tokens). Can we just compare the takerTokenAddress part out of this?

Reply:
That would be difficult because the takerAssetData is an unstructured byte array. It is not intended to be decoded outside of its destination proxy.

That said, this comparison should be fine now that we have agreed to remove the old ERC20 proxy. @abandeali1 - are you okay with leaving this comparison in?